### PR TITLE
Week 9 Updates

### PR DESCRIPTION
**Pull request by NovaSagittarii**
- @advilm https://github.com/Dragonlinae/Papereaven/pull/25
  - Implement Path2D based paths for moving platforms
- @NovaSagittarii https://github.com/Dragonlinae/Papereaven/pull/29
  - Center the pause menu
- @Kwok-Ho https://github.com/Dragonlinae/Papereaven/pull/30
  - Implement door environmental feature
- @Dragonlinae https://github.com/Dragonlinae/Papereaven/pull/26
  - Implement combo counter and timer GUI
- @Dragonlinae https://github.com/Dragonlinae/Papereaven/pull/27
  - Implement world selection and level navigation (interact with portal to move)
- @Dragonlinae https://github.com/Dragonlinae/Papereaven/pull/28
  - Implement tutorial world
- @Dragonlinae #35 
  - Implement spinning circle alignment puzzle.
- @SafeDuck #32 
  - Implement move box to press all buttons to advance puzzle.
- @Kwok-Ho #42 
  - Implement dodging-heavy based level (*seemingly incompatible)
- @AnPhu-D #40 
  - Implement Level 2
- @NovaSagittarii #41 
  - Implement slightly better player model flipping (still problematic)

### CODE REVIEW

File: character_controller_test.tscn [hunk 1]
```diff
@@ -1733,19 +1733,19 @@ script = ExtResource("3_tib08")
...
 [node name="CollisionShape2D" type="CollisionShape2D" parent="Hurtbox"]
 shape = SubResource("RectangleShape2D_sbxtj")
 
 [node name="HurtboxState" type="Node" parent="Hurtbox"]
 script = ExtResource("4_1c3yk")
 
+[node name="Sprite2D" type="Sprite2D" parent="Hurtbox"]
+position = Vector2(1.41799, -0.401596)
+scale = Vector2(0.285608, 0.28652)
+texture = ExtResource("4_380wk")
```
Comment (NovaSagittarii) [inline character_controller_test.tscn:1748]: for some reason, the newer one has named this resource to `7_dsikv`

**Review by Dragonlinae — APPROVED** [review]
Animation still a bit clunky, but other than that it looks good to go.

**Comment by NovaSagittarii** [comment]
the animations are so bugged 😔 🙏

**Review by Dragonlinae — COMMENTED** [review]
New change approved.

**Review by Dragonlinae — APPROVED** [review]

**Pull request merged**